File: tests/cases-py2many/demorgan.jl

function demorgan(a::Bool, b::Bool)::Bool
    return (a && b) == !(!(a) || !(b))
end

@assert(demorgan(true, true))
@assert(demorgan(true, false))
@assert(demorgan(false, true))
@assert(demorgan(false, false))
println("OK")

File: tests/cases/demorgan.py
#!/usr/bin/env python3

from py2many.smt import check_sat


def demorgan(a: bool, b: bool) -> bool:
    (a and b) == (not ((not a) or (not b)))


a: Bool
b: Bool

assert not demorgan(a, b)
check_sat()
